### PR TITLE
Implements lag correction for oiling robots.

### PR DESCRIPTION
**Pull request by vode-code**
Implements Mult to robot_oil.

## About the PR 
This PR moves the process_oil proc to the /datum/lifeprocess/robot_oil, and implements Mult lag correction to robot_oil.

## Why's this needed? 
Lag correction is good.

## Changelog

```
(u)Chickenish:
(+)The time that the oil buff for cyborgs lasts is no longer increased by lag.
```

### CODE REVIEW

File: code/mob/living/life/Life.dm [hunk 1]
```diff
@@ -331,7 +332,7 @@
 
 	process_killswitch()
 	process_locks()
-	process_oil()
+//	process_oil() moved to life process in robot_oil.dm
```
Comment (ZeWaka) [inline code/mob/living/life/Life.dm:335]:
```suggestion
```

File: code/mob/living/silicon/robot.dm [hunk 2]
```diff
@@ -2445,13 +2445,7 @@
...
 		src.oil += amt
 
-	proc/process_oil()
-		src.oil -= 1
-		if (oil <= 0)
-			oil = 0
-			src.remove_stun_resist_mod("robot_oil", 25)
-			REMOVE_MOVEMENT_MODIFIER(src, /datum/movement_modifier/robot_oil, "oil")
-
+//process_oil has been moved to /datum/lifeprocess/robot_oil in robot_oil.dm
```
Comment (ZeWaka) [inline code/mob/living/silicon/robot.dm:2448]:
```suggestion
```

File: code/mob/living/silicon/robot_oil.dm [hunk 3]
```diff
@@ -0,0 +1,11 @@
+/datum/lifeprocess/robot_oil
+	//moved from robot.dm proc/process_oil to this life process
```
Comment (ZeWaka) [inline code/mob/living/silicon/robot_oil.dm:2]:
```suggestion

```

File: code/mob/living/silicon/robot_oil.dm [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+	process(var/datum/gas_mixture/environment)
+		if(robot_owner)
+			var/mult = get_multiplier()
+			robot_owner.oil -= 1*mult
```
Comment (ZeWaka) [inline code/mob/living/silicon/robot_oil.dm:6]: why multiply 1 by it? useless

Comment (vode-code) [reply code/mob/living/silicon/robot_oil.dm:6]: right, wasn't thinking about that at the time, was trying to get everything around it to work.

File: code/mob/living/silicon/robot_oil.dm [hunk 5]
```diff
@@ -0,0 +1,10 @@
+/datum/lifeprocess/robot_oil
+	process(var/datum/gas_mixture/environment)
+		if(robot_owner)
+			var/mult = get_multiplier()
+			robot_owner.oil -= mult
```
Comment (ZeWaka) [inline code/mob/living/silicon/robot_oil.dm:5]:
```suggestion
			robot_owner.oil -= get_multiplier()
```

**Review by ZeWaka — APPROVED** [review]

**Comment by vode-code** [comment]
The current implementation in this PR would make one application of an oil can last thirty seconds, for fifteen units of oil out of the sixty units contained in the oil can.
This should probably be increased.

**Comment by vode-code** [comment]
Just realized chem borgs have beakers to splash themselves with, while I was presuming they would have to get a human to assist.
An increase in length no longer appears to be a good idea.

File: code/mob/living/silicon/robot_oil.dm [hunk 6]
```diff
@@ -0,0 +1,9 @@
+/datum/lifeprocess/robot_oil
+	process(var/datum/gas_mixture/environment)
+		if(robot_owner)
+			robot_owner.oil -= get_multiplier()
+			if (robot_owner.oil <= 0)
```
Comment (ZeWaka) [inline code/mob/living/silicon/robot_oil.dm:5]: i don't like how this subtracts even if there's no oil

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/mob/living/silicon/robot_oil.dm [hunk 7]
```diff
@@ -0,0 +1,10 @@
+/datum/lifeprocess/robot_oil
+	process(var/datum/gas_mixture/environment)
+		if(robot_owner)
+			if(robot_owner.oil > 0)
+				robot_owner.oil -= get_multiplier()
+				if (robot_owner.oil <= 0)
```
Comment (ZeWaka) [inline code/mob/living/silicon/robot_oil.dm:6]: this is unnecessary now

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**